File: PHP/selection_sort.php
<?php

class SelectionSort
{
    public function findSmallest(Array $arr)
    {
        $smallest = $arr[0];
        $smallest_index = 0;

        for ($i = 1; $i <= count($arr); $i++) {
            if ($arr[$i] < $smallest) {
                $smallest = $arr[$i];
                $smallest_index = $i;
            }
        }

        return $smallest_index;
    }

    public function selectionSort(Array $arr)
    {
        $newArr = [];

        for ($i = 1; $i <= count($arr); $i++) {
            $smallest = findSmallest($arr);
            $newArr = array_push(array_pop($smallest));
        }

        return $newArr;
    }
}

File: Golang/selection_sort.go
package selection_sort

func selection_sort(items []int) {
    var n = len(items)
    for i := 0; i < n; i++ {
        var minIdx = i
        for j := i; j < n; j++ {
            if items[j] < items[minIdx] {
                minIdx = j
            }
        }
        items[i], items[minIdx] = items[minIdx], items[i]
    }
} 